src/lib/draw-and-pick.js

```diff
@@ -95,16 +96,18 @@ export function pickLayers(gl, {
 
   } else {
     // For all other events, run picking process normally.
-    const pickInfo = pickFromBuffer(gl, {
+    const pickResult = pickFromBuffer(gl, {
       layers,
       pickingFBO,
       deviceX,
-      deviceY
+      deviceY,
+      deviceRadius
     });
-    pickedColor = pickInfo.pickedColor;
-    pickedLayer = pickInfo.pickedLayer;
-    pickedObjectIndex = pickInfo.pickedObjectIndex;
-    affectedLayers = pickInfo.affectedLayers;
+
+    pickedColor = pickResult.color;
+    pickedLayer = pickResult.layer;
+    pickedObjectIndex = pickResult.objectIndex;
+    affectedLayers = pickedLayer ? [pickedLayer] : [];
```

Comment:
The only logic that has been moved out is `affectedLayers`. `pickFromBuffer()` and `pickFromBufferByBoundingBox()` now return similarly shaped objects.